src/Microsoft.Tye.Core/Serialization/YamlConfigApplicationHelpers.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Tye.ConfigModel;
+using YamlDotNet.RepresentationModel;
+
+namespace Tye.Serialization
+{
+    public static class YamlConfigApplicationHelpers
+    {
+
+        public static void HandleConfigApplication(YamlMappingNode? yamlMappingNode, ConfigApplication app)
+        {
+            foreach (var child in yamlMappingNode!.Children)
+            {
+                string? key;
+                switch (child.Key.NodeType)
+                {
+                    case YamlNodeType.Scalar:
+                        key = (child.Key as YamlScalarNode)!.Value;
```

Comment:
Don't combine an `as` cast with `!`. You want this to be a c-style cast so it will fail with a good error instead of a nullref.